netket/experimental/driver/vmc_ngd.py

```diff
@@ -0,0 +1,411 @@
+
+    The regularization parameter :math:`\lambda` is the `diag_shift` parameter of the driver, which can be
+    a scalar or a schedule.
+    The updates are then applied to the parameters using the `optimizer` which in general should be `optax.sgd`.
+
+    .. note:: Momentum
+
+        When `momentum` is used, this driver implements the SPRING optimizer in
+        `G.Goldshlager et Al. <https://arxiv.org/abs/2401.10190>`_
+        to accumulate previous updates for better approximation of the exact SR with
+        no significant performance penalty.
+
+        `momentum` is a number between [0,1] that specifies the damping factor of the previous updates, and
+        works similarly to the beta parameter of ADAM. Therefore a value of `momentum=0.9` corresponds to a
+        moderate damping factor of 0.1, and means that if subsequent moves are close, the steps might be
+        amplified by at most a factor of 10. A value of `momentum=0.99` corresponds to a damping factor of 0.01,
+        and means that if subsequent moves are close, the steps might be amplified by at most a factor of 100.
+
+        Values that empirically work are around 0.8.
+
+        While we have found empirically that in practice momentum can be beneficial in some simulations, we find
+        that the theoretical analysis of this parameter is still lacking, and would welcome some contributions
+        in this direction.
+
+    .. note:: Implementation details
+
+        The kernel-trick/NTK based implementation can run with both a direct calculation of the jacobian
+        (`on_the_fly=False`) or with a lazy evaluation of the NTK (`on_the_fly=True`). The latter is more
+        computationally efficient for networks that reuse the parameters many times for every forward pass
+        (convolutions, attention layers, but not dense layers...) and generally uses less memroy.
+
+        However, the on the fly implementation relies on some JAX compiler behaaviour, so it might at times
+        have worse performance. We suggest you check on your specific model.
+
+    .. note:: Citation
+
+        - Stochastic Reconfiguration was originally introduced in the QMC field by `Sorella <https://arxiv.org/abs/cond-mat/9803107>`_.
+          The method was later shown to be equivalent to the Natural Gradient Descent method introduced by
+          `Amari <https://ieeexplore.ieee.org/abstract/document/6790500/>`_ in the context of neural network training.
+
+        - The *kernel tricK* which makes NGD/SR feasible in the large-parameter count limit was originally introduced
+          to the field of NQS by `Chen & Heyl <https://arxiv.org/abs/2302.01941>`_ who called it `minSR`. However,
+          `Rende & Al <https://arxiv.org/abs/2310.05715>`_ delivered a more straightforward derivation in terms of
+          the Kernel trick. In order to not pick a side, we will however mention that the kernel trick for Natural Gradient Descent
```

Comment:
I would avoid this jovial remark and just add the citation (again mentioning it was for the real NGD)